src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -1093,17 +1095,20 @@ public TraverseContext(scoped in TraverseContext context, int index)
 
             public TraverseContext(
                 Span<byte> updatePath,
-                CappedArray<byte> updateValue,
+                in CappedArray<byte> updateValue,
                 bool isUpdate,
                 bool ignoreMissingDelete = true)
             {
                 UpdatePath = updatePath;
                 if (updateValue.IsNotNull && updateValue.Length == 0)
                 {
-                    updateValue = new CappedArray<byte>(null);
+                    UpdateValue = ref CappedArray<byte>.Null;
+                }
+                else
+                {
+                    UpdateValue = ref updateValue;
```

Comment:
why use 8 lines of code when you can do that in 1 or max 3? ;)

Reply:
Used Ternary op for you :)